aiohttp/web_middlewares.py

```diff
@@ -73,7 +73,7 @@ def normalize_path_middleware(
                 resolves, request = await _check_request_resolves(
                     request, path)
                 if resolves:
-                    raise redirect_class(request.path + query)
```

Comment:
I wonder if `request.path` can be replaced with `request.raw_path`

Reply:
Thats the same as `path`, no? request.raw_path is being used above to populate paths_to_check.

Reply:
No. It is the percent encoded *canonical* path representation.

Reply:
And thats exactly the issue I am solving here, and its passing all the tests so I dont get it.